habitat-lab/habitat/tasks/rearrange/sub_tasks/pick_sensors.py

```diff
@@ -149,6 +154,47 @@ def reset_metric(self, *args, episode, task, observations, **kwargs):
             **kwargs,
         )
 
+
+    @staticmethod
+    def cosine(v1, v2):
+        return np.clip(np.dot(v1, v2), -1.0, 1.0)
+
+
+    def get_camera_angle_reward(self, obj_pos):
+        """Calculates angle between gripper line-of-sight and given global position."""
+
+        # Get the camera transformation
+        cam_T = self.get_camera_transform()
+        # Get object location in camera frame
+        cam_obj_pos = cam_T.inverted().transform_point(obj_pos).normalized()
+
+        # Get angle between (normalized) location and the vector that the camera should
+        # look at
+        reward = self.cosine(cam_obj_pos, mn.Vector3(0, 1, 0))
```

Comment:
I see very similar logic in the gaze grasp action. Can we put this code to calculate camera angle into some common utils file?

Reply:
Done. Moved to relevant utils files.